src/domains/wallet/pages/ImportWallet/Ledger/LedgerAuthStep.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useLayoutEffect, useState } from "react";
+
+import { useTranslation } from "react-i18next";
+import { Image } from "@/app/components/Image";
+import { Header } from "@/app/components/Header";
+import { Spinner } from "@/app/components/Spinner";
+import { useLedgerContext } from "@/app/contexts";
+
+export const LedgerAuthStep = ({
+	onDeviceNotAvailable,
+	onDeviceAvailable,
+}: {
+	onDeviceNotAvailable: () => void;
+	onDeviceAvailable: () => void;
+}) => {
+	const { t } = useTranslation();
+	const [isLookingDevice, setIsLookingDevice] = useState(true);
+	const [isLedgerConnected, setIsLedgerConnected] = useState(false);
+
+	const { listenDevice, hasDeviceAvailable, isAwaitingConnection } = useLedgerContext();
+
+	useLayoutEffect(() => {
+		(async () => {
+			await listenDevice();
+			setIsLookingDevice(false);
+		})();
+	}, [setIsLookingDevice, listenDevice]);
+
+	useLayoutEffect(() => {
+		if (hasDeviceAvailable && !isLookingDevice) {
+			setIsLedgerConnected(true);
+
+			onDeviceAvailable();
+		}
+	}, [hasDeviceAvailable, isLookingDevice, onDeviceAvailable, setIsLedgerConnected]);
+
+	useLayoutEffect(() => {
+		if (!isAwaitingConnection && !hasDeviceAvailable && !isLookingDevice) {
+			setIsLedgerConnected(false);
+
+			onDeviceNotAvailable();
+		}
+	}, [isAwaitingConnection, hasDeviceAvailable, isLookingDevice, onDeviceNotAvailable, setIsLedgerConnected]);
+
```

Comment:
@alfonsobries the above code is just fine, nice work :100:  , was only thinking that we could achieve the same functionality using only the state from `useLedgerContext` instead of keeping an additional ledger connection state :hmhm:

Something like this:

```suggestion
	const { listenDevice, hasDeviceAvailable, error, resetConnectionState } = useLedgerContext();

	useEffect(() => {
		listenDevice();
	}, [])

	useEffect(() => {
		// User clicked cancel, or permission dialog was closed.
		if(error === "Access denied to use Ledger device") {
			// Make sure to clear errors.
			resetConnectionState();
			onDeviceNotAvailable();
		}
	}, [error])

	useEffect(() => {
		if(hasDeviceAvailable) {
			onDeviceAvailable();
		}
	}, [hasDeviceAvailable])


```
The `resetConnectionState` is implemented in https://github.com/PayvoHQ/wallet/pull/941 to make sure all errors are cleared before leaving this step.